File: php/sieve/Sieve.php
<?php declare(strict_types=1);

function sieve(int $limit): array
{
    $list = range($offset = 2, $limit);

    foreach ($list as $value) {
        for ($i = ($value * 2) - $offset; $i <= $limit; $i += $value) {
            unset($list[(int) $i]);
        }
    }

    return array_values($list);
}

File: python/sieve/sieve.py
def primes(limit):
    if limit < 2:
        return []

    r = []
    multiples = []
    for i in range(2, limit+1):
        if i not in multiples:
            r.append(i)

            j = i*2
            while j <= limit:
                if j not in multiples:
                    multiples.append(j)
                j += i
    return r
